base/collection.go

```diff
@@ -140,7 +140,13 @@ func GetGocbV2BucketFromCluster(cluster *gocb.Cluster, spec BucketSpec, waitUnti
 		clusterCompatMajorVersion: uint64(clusterCompatMajor),
 		clusterCompatMinorVersion: uint64(clusterCompatMinor),
 	}
+	uuid, err := getServerUUID(gocbv2Bucket)
+	if err != nil {
```

Comment:
I don't think existing usages of getServerUUID are fatal on failed retrieval.  I'm not sure it needs to be in this case, but if it does, I suspect getServerUUID needs to be made more robust (to handle transient failures).

Reply:
Added a retry loop and we now only call it in one place.